crates/bevy_ecs/src/query/state.rs

```diff
@@ -583,6 +583,16 @@ impl<D: QueryData, F: QueryFilter> QueryState<D, F> {
             {
                 access.add_component_write(archetype_component_id);
             }
+            if self.component_access.access.has_resource_read(component_id) {
```

Comment:
Oh right, this is an existing soundness bug. Re-adding the tag.

Reply:
These changes have been removed in favor of @chescock's.